core/src/main/java/org/apache/carbondata/core/statusmanager/LoadMetadataDetails.java

```diff
@@ -468,4 +482,26 @@ public boolean isCarbonFormat() {
     return getFileFormat().equals(FileFormat.COLUMNAR_V3)
         || getFileFormat().equals(FileFormat.ROW_V1);
   }
+
+  /**
+   * Before writing table status file, call this to make the metadata smaller.
+   * It checks if fields are default value, then make it null so GSON does not write it
+   */
+  void removeUnnecessaryField() {
+    if (updateDeltaEndTimestamp == null || updateDeltaStartTimestamp.isEmpty()) {
+      updateDeltaEndTimestamp = null;
+    }
+    if (updateDeltaStartTimestamp == null || updateDeltaStartTimestamp.isEmpty()) {
+      updateDeltaStartTimestamp = null;
+    }
+    if (updateStatusFileName == null || updateStatusFileName.isEmpty()) {
+      updateStatusFileName = null;
+    }
+    if (visibility == null || visibility.equals("true")) {
+      visibility = null;
+    }
+    if (fileFormat == null || fileFormat.equals(FileFormat.COLUMNAR_V3.toString())) {
```

Comment:
It's better use (StringUtils.isEmpty(fileFormat) || fileFormat.equals(FileFormat.COLUMNAR_V3.toString()))

Reply:
fixed